lib/fdk/listener.rb

```diff
@@ -46,23 +51,41 @@ def listen(&block)
 
     def handle_request(fn_block:)
       local_socket = socket.accept
-      req = WEBrick::HTTPRequest.new(WEBrick::Config::HTTP)
-      resp = WEBrick::HTTPResponse.new(WEBrick::Config::HTTP)
+      req, resp = new_req_resp
       req.parse(local_socket)
       FDK.debug "got request #{req}"
+      log_frame_header(req.header)
       fn_block.call(req, resp)
       resp["Connection"] = "close" # we're not using keep alives sadly
       resp.send_response(local_socket)
       FDK.debug "sending resp  #{resp.status}, #{resp.header}"
       local_socket.close
     end
 
+    def new_req_resp
+      req = WEBrick::HTTPRequest.new(WEBrick::Config::HTTP)
+      resp = WEBrick::HTTPResponse.new(WEBrick::Config::HTTP)
+      [req, resp]
+    end
+
     def socket_path
       @socket_path ||= url[5..url.length]
     end
 
     def private_socket_path
       socket_path + ".private"
     end
+
+    def log_frame_header(headers)
+      return if @fn_logframe_name.nil? || @fn_logframe_hdr.nil?
```

Comment:
maybe same thing here, re: empty (empty string)

Reply:
Done. Had to refactor it out into a separate method to keep rubocop happy (else `Metrics/CyclomaticComplexity: Cyclomatic complexity for log_frame_header is too high`)